### PR TITLE
Fix missing customApiRegistryPromise param for test connection

### PR DESCRIPTION
**Pull request by BionIT**
### Description

This change adds custom API registry to the test connection route

### Issues Resolved
Fixes https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5937

## Screenshot

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/17714150/217dc0bd-885a-4c54-82b8-701ad7ca452e

## Testing the changes

The following steps were followed:

enabled data source plugin
click test connection and should be successful

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -96,6 +96,9 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - [BUG] Remove duplicate sample data as id 90943e30-9a47-11e8-b64d-95841ca0b247 ([5668](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5668))
 - [BUG][Multiple Datasource] Fix datasource testing connection unexpectedly passed with wrong endpoint [#5663](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5663)
 - [Table Visualization] Fix filter action buttons for split table aggregations ([#5619](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5619))
+- [BUG][Discover] Allow saved sort from search embeddable to load in Dashboard ([#5934](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5934))
```
Comment (bandinib-amzn) [inline CHANGELOG.md:99]: Nit: I think this got added mistakenly. This is not related to your change. Can you remove it?

Comment (BionIT) [reply CHANGELOG.md]: Thanks! This got in after merge from main, removed in new commit

**Review by bandinib-amzn — APPROVED** [review]

**Comment by Flyingliuhub** [comment]
@BionIT we probably need to add label for the backport 2.x 2.13, and may need to 2.12 as well

File: src/plugins/data_source/server/plugin.ts [hunk 2]
```diff
@@ -129,7 +129,8 @@ export class DataSourcePlugin implements Plugin<DataSourcePluginSetup, DataSourc
       router,
       dataSourceService,
       cryptographyServiceSetup,
-      authRegistryPromise
+      authRegistryPromise,
+      customApiSchemaRegistryPromise
```
Comment (Flyingliuhub) [inline src/plugins/data_source/server/plugin.ts:133]: do we know how this got missed at the first place?

**Comment by BionIT** [comment]
> @BionIT we probably need to add label for the backport 2.x 2.13, and may need to 2.12 as well

custom API registry is new and not released in 2.12

File: src/plugins/data_source/server/plugin.ts (hunk 2 above)
Comment (BionIT) [reply src/plugins/data_source/server/plugin.ts:133]: We need to add more tests for MDS to capture things like this, and I have added the only test for the test connection router. Can we fix this issue first then add more tests?

**Review by Flyingliuhub — APPROVED** [review]

**Comment by manasvinibs** [comment]
Manual Backport https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5971
